theSrc/scripts/PlotlyChartElements.js

```diff
@@ -0,0 +1,253 @@
+    let normZ
+    let marker_opacity = config.transparency
+    if (Array.isArray(data.Z)) {
+        const maxZ = _.max(data.Z)
+        normZ = LegendUtils.normalizeZValues(data.Z, maxZ).map(z => 2 * LegendUtils.normalizedZtoRadius(config.pointRadius, z))
+        if (marker_opacity === null) marker_opacity = 0.4
+        const z_title = config.zTitle.length === 0 ? '' : config.zTitle + ': '
+        tooltips = indices.map(i => `${tooltips[i]}<br>${z_title}${data.Z[i]}`)
+    }
+    if (marker_opacity === null) marker_opacity = 1.0
+    const plot_data = []
+    if (!Array.isArray(data.group)) {
+        const marker_size = data.Z === undefined ? config.pointRadius * 2 : normZ
+        plot_data.push({
+            x: data.X,
+            y: data.Y,
+            name: '',
+            text: tooltips,
+            hoverinfo: 'name+text',
+            hoverlabel: { font: { color: autoFontColor(config.colors[0]) } },
+            type: 'scatter',
+            mode: 'markers',
+            marker: {
+                color: config.colors[0],
+                size: marker_size,
+                sizemode: 'diameter',
+                opacity: marker_opacity,
+                outlinewidth: 0
+            },
+            cliponaxis: 'false',
+        })
+    } else {
+        const indices_by_group = _.groupBy(indices, i => data.group[i])
+        const group_names = Object.keys(indices_by_group)
+        for (let g = 0; g < group_names.length; g++) {
+            const gname = group_names[g]
+            const marker_size = normZ === undefined
+                ? config.pointRadius * 2
+                : _.at(normZ, indices_by_group[gname])
+            plot_data.push({
+                x: _.at(data.X, indices_by_group[gname]),
+                y: _.at(data.Y, indices_by_group[gname]),
+                text: _.at(tooltips, indices_by_group[gname]),
+                hoverinfo: 'name+text',
+                hoverlabel: { font: { color: autoFontColor(config.colors[g % config.colors.length]) } },
+                name: gname,
+                type: 'scatter',
+                mode: 'markers',
+                marker: {
+                    color: config.colors[g % config.colors.length],
+                    size: marker_size,
+                    sizemode: 'diameter',
+                    opacity: marker_opacity,
+                    line: { width: 0 },
+                },
+                cliponaxis: 'false',
+            })
+        }
+    }
+    return plot_data
+}
+
+function createPlotlyLayout (config) {
+    const plot_layout = {
+        xaxis: {
+            title: {
+                text: config.xTitle,
+                font: {
+                    family: config.xTitleFontFamily,
+                    color: config.xTitleFontColor,
+                    size: config.xTitleFontSize
+                }
+            },
+            color: '#444',
+            showgrid: config.grid,
+            gridcolor: '#eee',
```

Comment:
Move to a constant

Reply:
I've made this a variable that can set in buildConfig in [1a634a8](https://github.com/Displayr/rhtmlCombinedScatter/pull/7/commits/1a634a8d45572edebe6a5a52447ba3688e8456f0)